python/lsst/pipe/tasks/diff_matched_tract_catalog.py

```diff
@@ -664,247 +667,288 @@ def run(
-
-                        # e.g. if order is ugrizy, first color will be u - g
-                        cat_target[column_color_temp_model] = mag_prev[idx_model] - mag_model[idx_model]
-
-                        # Sum (in quadrature, and admittedly sketchy for faint fluxes) magnitude errors
-                        cat_target[column_color_err_temp_model] = 2.5/np.log(10)*np.hypot(
-                            flux_err_frac[idx_model], flux_err_frac_prev[idx_model])
-                        columns_target[f'color_{band_prev}_m_{band}_{column_target_flux}'] = (
-                            column_color_temp,
-                            column_color_temp_model,
-                            column_color_err_temp_model,
-                            False,
-                        )
-
-                    for idx_bin, (mag_lo, mag_hi) in enumerate(bins_mag):
-                        row = data[idx_bin]
-                        # Reference sources only need to be counted once
-                        if idx_model == 0:
-                            select_ref_bin = select_ref_bins[idx_bin]
-                        select_target_bin = select_target & (mag_target > mag_lo) & (mag_target < mag_hi)
-
-                        for sourcetype in SourceType:
-                            sourcetype_info = sourcetype.value
-                            is_extended = sourcetype_info.is_extended
-                            # Counts filtered by match selection and magnitude bin
-                            select_ref_sub = select_ref_bin.copy()
-                            select_target_sub = select_target_bin.copy()
-                            if is_extended is not None:
-                                is_extended_ref = (extended_ref == is_extended)
-                                select_ref_sub &= is_extended_ref
+        if config.include_unmatched:
+            # Create an unmatched table with the same schema as the matched one
+            # ... but only for objects with no matches (for completeness/purity)
+            # and that were selected for matching (or inclusion via config)
+            cat_right = apTab.Table.from_pandas(cat_ref[~matched_ref & select_ref].reset_index(drop=False))
+            cat_right.rename_columns(cat_right.colnames, [f"refcat_{col}" for col in cat_right.colnames])
+            match_row_target = catalog_match_target['match_row'].values
+            cat_left = cat_target[~(match_row_target >= 0) & select_target].reset_index(
+                drop=not has_index_left)
+            # This may be slower than pandas but will, for example, create
+            # masked columns for booleans, which pandas does not support.
+            # See https://github.com/pandas-dev/pandas/issues/46662
+            cat_unmatched = apTab.vstack([apTab.Table.from_pandas(cat_left), cat_right])
+
+        if config.output_statistics:
+            # Slightly smelly hack for when a column (like distance) is already relative to truth
```

Comment:
If this is smelly, what's the better way to do this? I know this isn't really a change on this PR, as it existed prior, but keen to hear thoughts on how we can make this operation less smelly in the future.

Reply:
@taranu :point_up:

Reply:
Update: it seems as though all of this `if` block will be removed at some future point, so I'm not so interested any longer in addressing my question above.

Reply:
Yeah, I'd rather deprecate the whole thing than spend more energy on fixes like this.